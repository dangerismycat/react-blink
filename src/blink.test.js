import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Blink, { toggleBlinkProp } from './blink';

configure({ adapter: new Adapter() });


describe('Blink', () => {
  describe('works with an interval prop', () => {
    const props = { interval: 1000 };

    const wrapper = mount(<Blink {...props} />);

    it('renders without blowing up', () => {
      expect(wrapper.exists()).toBeTruthy();
    });

    it('matches previous snapshot', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('works without an interval prop', () => {
    const wrapper = mount(<Blink />);

    it('renders without blowing up', () => {
      expect(wrapper.exists()).toBeTruthy();
    });

    it('matches previous snapshot', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});


describe('toggleBlinkProp', () => {
  const testState = { on: false };

  it('flips the state boolean correctly', () => {
    expect(toggleBlinkProp(testState).on).toBeTruthy();
  });

  it('handles multiple consecutive flips correctly', () => {
    const state = toggleBlinkProp(toggleBlinkProp(toggleBlinkProp(toggleBlinkProp(testState))));
    expect(state.on).toBeFalsy();
  });
});
