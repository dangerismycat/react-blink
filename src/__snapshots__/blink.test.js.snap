// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blink works with an interval prop matches previous snapshot 1`] = `
<Blink
  interval={1000}
>
  <div
    style={
      Object {
        "opacity": 1,
      }
    }
  />
</Blink>
`;

exports[`Blink works without an interval prop matches previous snapshot 1`] = `
<Blink>
  <div
    style={
      Object {
        "opacity": 1,
      }
    }
  />
</Blink>
`;
